// @fast-crud/fast-crud/dist/fast-crud.mjs
import { A as F, al as u, C as l, D as i, F as n, a6 as m, ah as c, N as p, Z as C, a8 as d, K as D, H as b, X as g, ag as f, ad as y, ac as T, ae as x, a9 as S, aa as h, af as L, ai as P, a0 as E, $ as I, T as R, V as O, U as V, _ as v, W as A, P as M, R as U, Q as k, S as w, J as B, I as H, O as W, E as j, M as G, a1 as K, a2 as q, a4 as z, a3 as J, L as N, Y as Q, aj as X, a5 as Y, a7 as Z, ab as _, G as $, ak as ss, h as as, p as es, f as os, y as ts, F as rs, k as Fs, ao as us, t as ls, q as is, B as ns, ap as ms, am as cs, an as ps, z as Cs, r as ds, s as Ds, v as bs, i as gs, b as fs, j as ys, l as Ts, w as xs, n as Ss, x as hs, d as Ls, e as Ps, a as Es, o as Is, m as Rs, u as Os } from "./index-bae3872a.mjs";
export * from "@fast-crud/ui-interface";
import { uiContext as As, useUi as Ms } from "@fast-crud/ui-interface";
import "vue";
import "lodash-es";
import "dayjs";
import "vue-router";
export {
  F as AsyncComputeValue,
  u as ColumnsFilterProvideKey,
  l as ComputeValue,
  i as Dict,
  n as FastCrud,
  m as FsActionbar,
  c as FsBox,
  p as FsButton,
  C as FsCell,
  d as FsColumnsFilterLayoutDefault,
  D as FsComponentRender,
  b as FsContainer,
  g as FsCrud,
  f as FsDateFormat,
  y as FsDictCascader,
  T as FsDictCascaderFormat,
  x as FsDictCheckbox,
  S as FsDictRadio,
  h as FsDictSelect,
  L as FsDictSwitch,
  P as FsDictTree,
  E as FsEditable,
  I as FsEditableCell,
  R as FsForm,
  O as FsFormHelper,
  V as FsFormItem,
  v as FsFormProvider,
  A as FsFormWrapper,
  M as FsIcon,
  U as FsIconSvg,
  k as FsIconify,
  w as FsLabel,
  B as FsLayoutCard,
  H as FsLayoutDefault,
  W as FsLoading,
  j as FsPage,
  G as FsRender,
  K as FsRowHandle,
  q as FsSearch,
  z as FsSearchLayoutDefault,
  J as FsSearchV1,
  N as FsSlotRender,
  Q as FsTable,
  X as FsTableSelect,
  Y as FsTabsFilter,
  Z as FsToolbar,
  _ as FsValuesFormat,
  $ as GlobalConfig,
  ss as SetFormDataOptions,
  as as asyncCompute,
  es as buildTableColumnsFlatMap,
  os as compute,
  ts as crudOptionsPlugins,
  rs as default,
  Fs as dict,
  us as exportTable,
  ls as forEachColumns,
  is as forEachTableColumns,
  ns as getCrudOptionsPlugin,
  ms as importTable,
  cs as loadFsExportUtil,
  ps as loadFsImportUtil,
  Cs as registerCrudOptionsPlugin,
  ds as registerMergeColumnPlugin,
  Ds as setLogger,
  As as uiContext,
  bs as useColumns,
  gs as useCompute,
  fs as useCrud,
  ys as useDict,
  Ts as useDictDefine,
  xs as useDrag,
  Ss as useExpose,
  hs as useFormWrapper,
  Ls as useFs,
  Ps as useFsAsync,
  Es as useI18n,
  Is as useMerge,
  Rs as useTypes,
  Ms as useUi,
  Os as utils
};
